### DISCUSSION TITLE
API Hook - warnings

### DISCUSSION THREAD

**Discussion by Meldiron** [directus/directus#5159] [Feature Requests]
Currently, we can throw exceptions in `items.update.before` API hooks that results in error when trying to save changes. This is just fine for most cases, but sometimes we simply need warnings. Warning means, it is possible you really wanted to do this change, but not likely. We are all humans and we all do mistakes...

For example, wouldn't it be nice to see alert `You entered price 4999€. Do you really want to save with this price or did you mean to type 49.99€`? Yes, we might have a product for 5000€, but I saw so many editors do this kind of simple mistake because they copied the price from Excel and somehow lost the decimal separator.

Solution for this should not be too complex... We could implement new exception type `WarningException` and just like with any other exception, provide some message.

When this specific exception arrives to frontend, instead of showing error alert and `Dismiss` button, we show warning alert with options `Continue editing` and `Save anyway`.

![image](https://user-images.githubusercontent.com/19310830/115369018-f4a5c300-a1c7-11eb-99a2-0a9f759b65ce.png)

**Comment by rijkvanzanten** [comment]
> Solution for this should not be too complex... We could implement new exception type WarningException and just like with any other exception, provide some message.

Yes and no.. The main problem here is that there needs to be an additional way for the request to indicate that the specific warning is "allowed" and that the API should proceed. So not only should there be a sort of specific code for this exact warning, there should be an additional flag in the request payload that includes the "accepted warnings" 🤔

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
